akka-remote/src/test/scala/akka/remote/serialization/AllowJavaSerializationOffSpec.scala

```diff
@@ -0,0 +1,132 @@
+
+object AllowJavaSerializationOffSpec {
+
+  val testSerializer = new TestSerializer
+
+  val serializationSettings = SerializationSetup { _ ⇒
+    List(
+      SerializerDetails("test", testSerializer, List(classOf[ProgrammaticDummy])))
+  }
+  val bootstrapSettings = BootstrapSetup(None, Some(ConfigFactory.parseString("""
+    akka {
+      actor {
+        serialize-messages = off
+
+        serialization-bindings {
+          "akka.serialization.ConfigurationDummy" = test
+        }
+      }
+    }
+    """)), None)
+  val actorSystemSettings = ActorSystemSetup(bootstrapSettings, serializationSettings)
+
+  val noJavaSerializationSystem = ActorSystem("AllowJavaSerializationOffSpec" + "NoJavaSerialization", ConfigFactory.parseString(
+    """
+    akka {
+      actor {
+        allow-java-serialization = off
+      }
+    }
+    """.stripMargin))
+  val noJavaSerializer = new DisabledJavaSerializer(noJavaSerializationSystem.asInstanceOf[ExtendedActorSystem])
+
+}
+
+class AllowJavaSerializationOffSpec extends AkkaSpec(
+  ActorSystem("AllowJavaSerializationOffSpec", AllowJavaSerializationOffSpec.actorSystemSettings)) {
+
+  import AllowJavaSerializationOffSpec._
+
+  // This is a weird edge case, someone creating a JavaSerializer manually and using it in a system means
+  // that they'd need a different actor system to be able to create it... someone MAY pick a system with
+  // allow-java-serialization=on to create the SerializationSetup and use that SerializationSetup
+  // in another system with allow-java-serialization=off
+  val addedJavaSerializationSettings = SerializationSetup { _ ⇒
+    List(
+      SerializerDetails("test", testSerializer, List(classOf[ProgrammaticDummy])),
+      SerializerDetails("java-manual", new JavaSerializer(system.asInstanceOf[ExtendedActorSystem]), List(classOf[ProgrammaticJavaDummy])))
+  }
+  val addedJavaSerializationProgramaticallyButDisabledSettings = BootstrapSetup(None, Some(ConfigFactory.parseString("""
+    akka {
+      loglevel = debug
+      actor {
+        allow-java-serialization = off
+      }
+    }
+    """)), None)
+
+  val dontAllowJavaSystem =
+    ActorSystem("addedJavaSerializationSystem", ActorSystemSetup(addedJavaSerializationProgramaticallyButDisabledSettings, addedJavaSerializationSettings))
+
+  "Disabling java serialization" should {
+
+    "throw if passed system to JavaSerializer has allow-java-serialization = off" in {
+      intercept[DisabledJavaSerializer.JavaSerializationException] {
+        new JavaSerializer(noJavaSerializationSystem.asInstanceOf[ExtendedActorSystem])
+      }.getMessage should include("akka.actor.allow-java-serialization = off")
+
+      intercept[DisabledJavaSerializer.JavaSerializationException] {
+        SerializationExtension(dontAllowJavaSystem).findSerializerFor(new ProgrammaticJavaDummy).toBinary(new ProgrammaticJavaDummy)
+      }
+    }
+
+    "enable additional-serialization-bindings" in {
```

Comment:
the test config should probably use `enable-additional-serialization-bindings=off` to test that the `allow-java-serialization = off` overrides that